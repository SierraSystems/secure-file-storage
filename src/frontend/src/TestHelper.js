import renderer from "react-test-renderer";

export default function testBasicSnapshot(component) {
  const model = renderer.create(component);
  const tree = model.toJSON();

  expect(tree).toMatchSnapshot();
}
