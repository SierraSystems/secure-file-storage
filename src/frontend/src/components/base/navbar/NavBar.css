.links {
  width: 100px;
  color: #ffffff !important;
  cursor: pointer;
}

.blue {
  background-color: #2d3047 !important;
  min-height: 210px;
}

.main-heading {
  color: #ffffff !important;
  padding-left: 40px;
  padding-top: 20px;
  padding-bottom: 20px;
  font-size: 35px;
}
