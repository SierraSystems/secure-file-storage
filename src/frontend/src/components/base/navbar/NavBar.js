import React from "react";
import "./NavBar.css";
import "bootstrap/dist/css/bootstrap.css";

export const NavBar = () => (
  <div className="blue spacing">
    <nav className="navbar navbar-expand-md sticky-top">
      <div className="container-fluid">
        <a className="navbar-brand" href="https://ca.nttdata.com/en/">
          <img
            src={`${process.env.PUBLIC_URL}/images/s3.svg`}
            height="40px"
            width="55px"
            alt="s3 logo"
          />
        </a>
      </div>
    </nav>
    <p className="main-heading">Secure File Upload & Storage</p>
  </div>
);
