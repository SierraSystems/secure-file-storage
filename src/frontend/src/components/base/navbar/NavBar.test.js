import React from "react";
import NavBar from "./NavBar";
import { generateSnapshotTree } from "../../../TestHelper";

describe("NavBar", () => {
  test("Component renders as expected", () => {
    expect(generateSnapshotTree(<NavBar />)).toMatchSnapshot();
  });
});
