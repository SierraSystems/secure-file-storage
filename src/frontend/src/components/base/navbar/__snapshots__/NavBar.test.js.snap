// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component renders as expected 1`] = `
<div
  className="blue spacing"
>
  <nav
    className="navbar navbar-expand-md sticky-top"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="https://ca.nttdata.com/en/"
      >
        <img
          alt="s3 logo"
          height="40px"
          src="undefined/images/s3.svg"
          width="55px"
        />
      </a>
    </div>
    <a
      className="links"
      href="https://ca.nttdata.com/en/"
    >
      About Us
    </a>
    <a
      className="pl-4 links"
      href="https://github.com/SierraSystems/secure-file-storage"
    >
      GitHub
    </a>
  </nav>
  <p
    className="main-heading"
  >
    Secure File Upload & Storage
  </p>
</div>
`;
